packages/frontend/app/nav/navigation.ts

```diff
@@ -0,0 +1,242 @@
+        }
+    })[0];
+
+    return {
+        children: [
+            editionLinks.newsPillar,
+            editionLinks.opinionPillar,
+            editionLinks.sportPillar,
+            editionLinks.culturePillar,
+            editionLinks.lifestylePillar,
+        ],
+        otherLinks: editionLinks.otherLinks,
+        brandExtensions: editionLinks.brandExtensions,
+    };
+};
+
+const find = (
+    links: LinkType[],
+    p: (a: LinkType) => boolean,
+): LinkType | undefined => {
+    if (links.length === 0) {
+        return undefined;
+    }
+
+    const [head, ...tail] = links;
+    return p(head) ? head : find(tail.concat(head.children), p);
+};
+
+const getChildrenFromOtherEditions = (
+    edition: string,
+    navData: any,
+): LinkType[] => {
+    return ['uk', 'au', 'us', 'international']
+        .filter(e => e !== edition)
+        .flatMap(e => navData[e].children)
+        .concat(navData.otherLinks);
+};
+
+const findDescendantByUrl = (
+    url: string,
+    edition: string,
+    pillars: LinkType[],
+    otherLinks: LinkType[],
+    navData: any,
+): LinkType | undefined => {
+    const hasUrl = (link: LinkType): boolean => link.url === url;
+    const found = find(pillars.concat(otherLinks), hasUrl);
+    return found
+        ? found
+        : find(getChildrenFromOtherEditions(edition, navData), hasUrl);
+};
+
+const findParent = (
+    currentNavLink: LinkType,
+    edition: string,
+    pillars: LinkType[],
+    otherLinks: LinkType[],
+    navData: any,
+): LinkType | undefined => {
+    // Football is currently in the News Pillar and the Sport pillar, however we don't want the parent to be News.
+    const isFootballInNews = (parentTitle: string): boolean =>
+        currentNavLink.title === 'Football' && parentTitle === 'News';
+
+    const isParent = (link: LinkType): boolean =>
+        link === currentNavLink ||
+        (link.children.includes(currentNavLink) &&
+            !isFootballInNews(link.title));
+
+    const found = find(pillars.concat(otherLinks), isParent);
+    return found
+        ? found
+        : find(getChildrenFromOtherEditions(edition, navData), isParent);
+};
+
+const getPillar = (
+    currentParent: LinkType | undefined,
+    edition: string,
+    pillars: LinkType[],
+    otherLinks: LinkType[],
+    navData: any,
+): LinkType | undefined => {
+    if (currentParent === undefined) {
+        return undefined;
+    }
+
+    if (otherLinks.includes(currentParent)) {
+        return undefined;
+    }
+
+    if (pillars.includes(currentParent)) {
+        return currentParent;
+    }
+
+    const found = findParent(
+        currentParent,
+        edition,
+        pillars,
+        otherLinks,
+        navData,
+    );
+    return found ? found : navData.uk.newsPillar;
+};
+
+const makeGuardianNavigation = (
+    navData: any,
+    edition: EditionLong,
+    currentUrl: string,
+    customSignPosting: NavItem | undefined,
+    readerRevenuePositions: ReaderRevenuePositions,
```

Comment:
Not directly about this PR, but I am not really sure why we have readerRevenuePositions as part of the nav. And also what the naming means here.